Source/WebCore/dom/Document.cpp

```diff
@@ -8906,12 +8906,16 @@ void Document::handlePopoverLightDismiss(const PointerEvent& event, Node& target
             auto isShowingAutoPopover = [](HTMLElement& element) -> bool {
                 return element.popoverState() == PopoverState::Auto && element.popoverData()->visibilityState() == PopoverVisibilityState::Showing;
             };
-            for (auto& element : lineageOfType<HTMLElement>(*startElement)) {
-                if (!clickedPopover && isShowingAutoPopover(element))
-                    clickedPopover = &element;
+            for (auto* element = startElement; element; element = element->parentElementInComposedTree()) {
```

Comment:
Unfortunate that we don’t have a `lineageOfType` function that works with the composed tree.

Reply:
Please also use RefPtr here.